pysumma/preprocessing/create_initial_condition.py

```diff
@@ -0,0 +1,152 @@
+from osgeo import gdal
+import pandas as pd
+import os
+import numpy as np
+import json
+import pkg_resources
+from .create_local_attribute import LocalAttributes
+
+PARAMETER = pkg_resources.resource_filename(
+        __name__, 'meta/hru_ic.json')
+with open(PARAMETER, 'r') as f:
+    hru_ic = json.load(f)
+
+PARAMETER = pkg_resources.resource_filename(
+        __name__, 'meta/mid_ic.json')
+with open(PARAMETER, 'r') as f:
+    mid_ic = json.load(f)
+
+ssurgo_soil_db = pkg_resources.resource_filename(
+        __name__, 'meta/ssurgo_soils_db.csv')
+
+class InitialCondition(object):
+
+    def __init__(self, filepath):
+        self.filepath = filepath
+
+    def tif_to_dataframe(self, path):
+        raster = gdal.Open(path)
+        rasterArray = raster.ReadAsArray()
+        df = pd.DataFrame(rasterArray)
+        return df
+
+    def tif_to_dataframe_one_col(self, path, col_name):
+        raster = gdal.Open(path)
+        rasterArray = raster.ReadAsArray()
+        df = pd.DataFrame(rasterArray)
+        one_col = pd.concat([pd.DataFrame(df.loc[:,[i]].values, columns=[col_name]) for i in range(len(df.columns))], ignore_index=True)
+        return one_col
+
+    def delete_number(self, df, number):
+        df.replace(number, np.nan, inplace=True)
+        df_new = df.dropna().astype('int64')
+        return df_new
+
+    def merge_mukey_soil_depth(self, df, soil_depth_df):
+        mukey_depth = df.merge(soil_depth_df, on='MUKEY')
+        return mukey_depth
+
+    def gru_hru_dim_initial_condition(self, gru_name, dir_grassdata):
+        SSURGO_soil_depth = pd.read_csv(ssurgo_soil_db)
+        la = LocalAttributes(dir_grassdata)
+        gru_mukey_1_col = la.tif_to_dataframe_one_col(os.path.join(dir_grassdata, gru_name +"_mukey.tif"), "MUKEY")
+        gru_mukey_1_col.replace(-2147483648, np.nan, inplace=True)
+        gru_mukey_1_col = gru_mukey_1_col.dropna().astype('int64')
+        gru_mukey_depth = gru_mukey_1_col.merge(SSURGO_soil_depth, on='MUKEY')
+        gru_mukey_layer = gru_mukey_depth[['nSoil']]
+        gru_mukey_layer['nSoil'].unique().max()
+        gru_initial_cond_hru = gru_mukey_layer.assign(**{'scalarv': hru_ic["scalarv"], 'scalarSnowAlbedo': hru_ic["scalarSnowAlbedo"], 
```

Comment:
There are a lot of really long lines in this PR, could you work to split some of it so line length is < 100 characters long?